kernel/src/main/scala/cats/kernel/instances/unit.scala

```diff
@@ -4,16 +4,18 @@ package instances
 package object unit extends UnitInstances
 
 trait UnitInstances {
-  implicit val catsKernelStdOrderForUnit: Order[Unit] =
+  implicit val catsKernelStdOrderForUnit: Order[Unit] with Hash[Unit] =
     new UnitOrder
 
   implicit val catsKernelStdAlgebraForUnit: BoundedSemilattice[Unit] with CommutativeGroup[Unit] =
     new UnitAlgebra
 }
 
-class UnitOrder extends Order[Unit] {
+class UnitOrder extends Order[Unit] with Hash[Unit] {
   def compare(x: Unit, y: Unit): Int = 0
 
+  def hash(x: Unit): Int = 0
```

Comment:
it might be better to use something other than `0` here. Maybe `"Unit".##` or something.

Reply:
I'd like to use `0` because in Scala, `().##` is 0. I want the default hash behavior the same as the JVM universal hash.